workspaces/ui-v2/src/entry-points/local/EnsureDaemonRunning.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { FC, useEffect, useState } from 'react';
+import { makeStyles } from '@material-ui/core';
+import { Client } from '@useoptic/cli-client';
+
+import { FullPageLoader } from '<src>/components';
+import * as SupportLinks from '<src>/constants/SupportLinks';
+
+export const EnsureDaemonRunning: FC = ({ children }) => {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    (async () => {
+      const cliClient = new Client('/api');
+      try {
+        await cliClient.health();
+        setLoading(false);
+      } catch (e) {
+        setError(true);
+        setLoading(false);
+      }
+    })();
+  }, []);
+
+  return loading ? (
+    <FullPageLoader title="loading" />
+  ) : error ? (
+    <CliDaemonUnreachableError />
+  ) : (
+    <>{children}</>
+  );
+};
+
+const CliDaemonUnreachableError: FC = () => {
+  const classes = useStyles();
+
+  return (
+    <div className={classes.errorContainer}>
+      <h2>Error reaching daemon</h2>
+      <p>
+        It appears the cli daemon is unreachable. This usually means that the
+        daemon crashed and needs to be restarted.
+      </p>
+      <p>
+        You can restart the daemon by running <code>api daemon:start</code> or
```

Comment:
I would say `api spec` or `api start` as opposed to daemon:start because spec/start etc also create a session so the page they are on will still probably work. If they do daemon:start the session will not exist and the page will not be able to reload properly. Essentially, it's not clear what you want to happen once they restart the daemon given they have this page open. Thoughts?

Reply:
Hmmm api spec seems to be better - I think there might be some work we want to do to change the daemon behavior such that you need to run api _abc_ to run the UI or daemon explicitly + redraw the boundaries of how we use this. It seems we've been having some mismatched UI versions hitting older running daemons (with old running code) - such that I think it's more explicit and more intuitive that to see the UI, you need to run `api spec` or something similar, and closing that will close both the UI and daemon

Something for outside of this PR - but I think that we more issues with mismatched versions if we don't address this